generic/tests/linux_stress.py

```diff
@@ -19,19 +22,41 @@ def run(test, params, env):
     :param env: Dictionary with test environment
     """
 
-    vm = env.get_vm(params['main_vm'])
-    vm.verify_alive()
-    stress = utils_test.VMStress(vm, 'stress', params)
-    stress.load_stress_tool()
-    stress_duration = int(params.get('stress_duration', 0))
-    # NOTE: stress_duration = 0 ONLY for some legacy test cases using
-    # autotest stress.control as their sub test.
-    # Please DO define stress_duration to make sure the clean action
-    # being performed, if your case can not be controlled by time,
-    # use utils_test.VMStress() directly
+    stress_duration = int(params.get("stress_duration", ""))
+    vms = env.get_all_vms()
+    up_time = {}
+    error = 0
+
+    for vm in vms:
+        try:
+            up_time[vm.name] = vm.uptime()
+            stress_server = utils_test.VMStress(vm, params)
+            stress_server.load_stress_tool()
+        except exceptions.TestError as err_msg:
+            error = 1
+            logging.error(err_msg)
 
     if stress_duration:
         time.sleep(stress_duration)
-        stress.unload_stress()
-        stress.clean()
-        vm.verify_kernel_crash()
+
+    for vm in vms:
+        try:
+            uptime = vm.uptime()
+            s_ping, o_ping = utils_test.ping(vm.get_address(), count=5, timeout=20)
+            if s_ping != 0:
+                error = 1
+                logging.error("%s seem to have gone out of network", vm.name)
+                continue
+            if up_time[vm.name] > uptime:
+                error = 1
+                logging.error("%s seem to have rebooted during the stress run", vm.name)
+            stress_server = utils_test.VMStress(vm, params)
+            stress_server.unload_stress()
```

Comment:
more efficient way is to reuse VMstress object created during load_stress_tool() here instead of recreating new object here.

Reply:
Done.